samples/observability-traces.js

```diff
@@ -0,0 +1,139 @@
+} = require('@opentelemetry/semantic-conventions');
+
+const resource = Resource.default().merge(
+  new Resource({
+    [SEMRESATTRS_SERVICE_NAME]: 'spanner-sample',
+    [SEMRESATTRS_SERVICE_VERSION]: 'v1.0.0', // The version of your app running.,
+  })
+);
+
+/*
+ * Uncomment these lines to debug OpenTelemetry.
+const {diag, DiagConsoleLogger, DiagLogLevel} = require('@opentelemetry/api');
+diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.ALL);
+*/
+
+// Create the Google Cloud Trace exporter for OpenTelemetry.
+const {
+  TraceExporter,
+} = require('@google-cloud/opentelemetry-cloud-trace-exporter');
+const exporter = new TraceExporter();
+
+// Optionally, you can enable gRPC instrumentation, by removing this guard.
+if (process.env.SPANNER_ENABLE_GRPC_INSTRUMENTATION === 'true') {
+  // eslint-disable-next-line n/no-extraneous-require
+  const {registerInstrumentations} = require('@opentelemetry/instrumentation');
+  const {GrpcInstrumentation} = require('@opentelemetry/instrumentation-grpc');
+  registerInstrumentations({
+    instrumentations: [new GrpcInstrumentation()],
+  });
+}
+
+async function main(
+  projectId = 'my-project-id',
+  instanceId = 'my-instance-id',
+  databaseId = 'my-project-id'
+) {
+  // Create the OpenTelemetry tracerProvider that the exporter shall be attached to.
+  const provider = new NodeTracerProvider({
+    // Trace every single request to ensure that we generate
+    // enough traffic for proper examination of traces.
+    sampler: new TraceIdRatioBasedSampler(1.0),
+    resource: resource,
+  });
+  provider.addSpanProcessor(new BatchSpanProcessor(exporter));
+
+  // Uncomment this to make it a global tracerProvider instead
+  // of passing it into SpannerOptions.observabilityOptions.
+  // provider.register();
+
+  // Create the Cloud Spanner Client.
+  const {Spanner} = require('@google-cloud/spanner');
+
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance-id';
+  // const databaseId = 'my-database-id';
+
+  const spanner = new Spanner({
+    projectId: projectId,
+    observabilityOptions: {
+      // Optional, but you could rather register the global tracerProvider.
+      tracerProvider: provider,
+      // This option can also be enabled by setting the environment
+      // variable `SPANNER_ENABLE_EXTENDED_TRACING=true`.
+      enableExtendedTracing: true,
+    },
+  });
+
+  // Acquire the database handle.
+  const instance = spanner.instance(instanceId);
+  const database = instance.database(databaseId);
+
+  try {
+    const query = {
+      sql: 'SELECT 1',
+    };
+    const [rows] = await database.run(query);
+    console.log(`Query: ${rows.length} found.`);
+    rows.forEach(row => console.log(row));
+  } finally {
+    spanner.close();
+  }
+
+  provider.forceFlush();
+
+  // This sleep gives ample time for the trace
```

Comment:
Do we really need sleep when we are doing explicit flush ? We were able to see spans without sleep if we add a flush

Reply:
We really do, the Google Cloud Trace Exporter takes a while to send content to the upstream as it has a background process to upload spans in batches to avoid huge charges from customers, even worse is the fact that the Cloud Trace Exporter doesn't have an explicit forceFlush for itself; on the metrics exporter has the explicit forceFlush :-(